`unobserveallproperties` operation - closes #58

benfrancis opened the pull request:
Closes #58.

This PR defines request and response messages for the unobserveallproperties operation, as discussed in in #42 and #58.

Similar open question to #86 regarding overlapping subscriptions.


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/benfrancis/web-thing-protocol/pull/88.html" title="Last updated on Sep 17, 2025, 4:58 PM UTC (028fb4f)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/web-thing-protocol/88/e9d70ae...benfrancis:028fb4f.html" title="Last updated on Sep 17, 2025, 4:58 PM UTC (028fb4f)">Diff</a>

hspaay reviewed (changes requested):
I request the following changes/clarifications:
1. unobserveallproperties is idempotent. It can be invoked multiple times with the same result.
2. unobserveallproperties removes all property registrations regardless of whether they have been registered with observeproperty, observemultipleproperties or observeallproperties. It is simply a bulk operations for convenience.

benfrancis commented:
Thanks @hspaay. Similar to PR #86 I have removed the error response and added an assertion to clarify that a Thing should still send a success response even if there were not active observation subscriptions.

Are you happy with this PR now?

hspaay reviewed (approved):
Yes, perfect, thank you!
